benchmarks/flat_map_vs_inject.rb

```diff
@@ -0,0 +1,24 @@
+require 'benchmark/ips'
+require 'rspec/core/flat_map'
+
+words = %w[ foo bar bazz big small medium large tiny less more good bad mediocre ]
+
+Benchmark.ips do |x|
+  x.report("flat_map") do
+    words.flat_map(&:codepoints)
+  end
+
+  x.report("inject") do
+    words.inject([]) { |a, w| a + w.codepoints }
```

Comment:
That's a good point.  I added a case for `concat` in the next commit.  Looks like `flat_map` is still faster, but the overhead of our `FlatMap.flat_map` compatibility layer (necessary to support 1.8.7 w/o monkey patching `flat_map` on enumerable) makes that essentially even with `inject`/`concat`.

I think the flat map version reads nicer and will make it easy to transition to raw `flat_map` when we drop support for 1.8.7 (probably in RSpec 4), so I decided to stick with it.